### PR TITLE
update apis for perf eval screen and user eval admin list

### PR DESCRIPTION
**Pull request by emmaacha**
Ticket ID: [Create API for Evaluations - Eval Admin List #83](https://github.com/nerubia/kh360-nodejs/issues/83)

FE PR: [create performance evaluation form and eval admin list #117 ](https://github.com/nerubia/kh360-react/pull/117)

### CODE REVIEW

File: src/controllers/admin/evaluation_template_contents_controller.ts [hunk 1]
```diff
@@ -3,11 +3,17 @@ import prisma from "../../utils/prisma"
 
 /**
  * List evaluation template contents based on provided filters.
- * @param req.query.evaluation_template_id - Filter by evaluation template id.
+ * @param req.query.evaluation_id - Filter by evaluation id.
  */
 export const index = async (req: Request, res: Response) => {
```
Comment (jatnerubia) [inline src/controllers/admin/evaluation_template_contents_controller.ts:8]: we should move this to user? 🤔

File: src/controllers/user/user_controller.ts [hunk 2]
```diff
@@ -28,12 +30,18 @@ export const getEvaluations = async (req: Request, res: Response) => {
           evaluation_administration_id as string
         ),
         evaluator_id: user.id,
-        for_evaluation: Boolean(parseInt(for_evaluation as string)),
```
Comment (jatnerubia) [inline src/controllers/user/user_controller.ts:31]: we should use `0` or `1` in the request then parse it

File: src/controllers/user/user_controller.ts [hunk 3]
```diff
@@ -88,12 +100,150 @@ export const getEvaluations = async (req: Request, res: Response) => {
       })
     )
 
+    const sortedEvaluations = evaluationDetails.sort((a, b) => {
```
Comment (jatnerubia) [inline src/controllers/user/user_controller.ts:103]: sort it on FE

**Review by Dev-Nino — APPROVED** [review]

File: src/controllers/user/user_controller.ts [hunk 4]
```diff
@@ -251,6 +432,45 @@ export const submitEvaluation = async (req: Request, res: Response) => {
       },
     })
 
+    const allEvaluationRatings = await prisma.evaluation_ratings.findMany({
```
Comment (jatnerubia) [inline src/controllers/user/user_controller.ts:435]: let's just check the comment in `submitEvaluation`

**Review by jatnerubia — CHANGES_REQUESTED** [review]
just a few changes ^^

File: src/routes/admin/users_route.ts [hunk 5]
```diff
@@ -6,4 +6,6 @@ const router = express.Router()
 router.get("/", index)
 router.get("/all", getAllUsers)
 
+router.get("/all", getAllUsers)
```
Comment (jatnerubia) [inline src/routes/admin/users_route.ts:9]: duplicate? 🤔

File: src/types/userType.ts
Comment (jatnerubia) [inline src/types/userType.ts:1]: do we still use this? 🤔

**Review by jatnerubia — CHANGES_REQUESTED** [review]
just a few changes ^^

File: src/types/userType.ts
Comment (emmaacha) [reply src/types/userType.ts:1]: Nice catch kuys, will remove, thanks!

**Review by jatnerubia — APPROVED** [review]
Nice

**Pull request merged**